### ISSUE TITLE
v0.20 - Array Index Out of Bounds on Midboss Spells

### ISSUE THREAD

**Issue by Sparen** [ExPorygon/Ephemeral-Unnatural-Balance#6]
As of a5c3cc79d087ec398f1b0155f290b171665ffcaf, playing Stage 2 via Stage practice crashes on mid boss Minoriko's second attack, which cites the following:

```
Array index out of bounds.
(配列のサイズを超えています)
Z:/Users/andrewfan/ssh/Ephemeral-Unnatural-Balance/System/Effect.dnh 
Effect.dnh line(行)=1588

↓
			
			let objtext3b = ObjText_Create;
```

**Comment by Sparen** [comment]
Oh yeah, and remember for the future that paths are case sensitive if you decide to use archives.

**Comment by ExPorygon** [comment]
Tell me if you were bombing or not during the spell, or if you were doing anything else unusual.

**Comment by Sparen** [comment]
I have no idea. I will attempt to replicate the bug.

**Comment by Sparen** [comment]
Confirmed:
Rin Satsuki, Normal Mode, no deaths or bombs during non spell. *I believe it defaults to Hard Mode
Kasen Ibaraki, Easy Mode, no deaths or bombs during non spell. *Defaults to Hard Mode

Due to the lack of any array indexing in Effect.dnh at line 1588, the problem is most likely elsewhere in the script that included the file.

Of course, I don't know enough about the library structures yet, but you're including the same file twice in the same script as well, and Danmakufu is not giving you a proper error for that. So I can't discern the real problem.

**Comment by Sparen** [comment]
Confirmed on Stage 3, Spell Practice, Easy Mode, Rin Satsuki

Player was already dead when the error occurred at the start of Aya's mid boss second attack.

**Comment by Sparen** [comment]
Update: This occurs when using Stage Practice from the package but did not occur when running Test Stage.dnh from the individual stage folders.

**Comment by ExPorygon** [comment]
Update: This error occurs for any spellcard that occurs during stage practice on any stage.

    if(IsCommonDataAreaExists("PlayState")) {
				if(GetAreaCommonData("PlayState","Mode","")!="Practice") {
					ObjText_SetText(objtext3b,rtos("00",GetAreaCommonData("History","Stage History",[])[SpellID][1]));
				}
				else {
					ObjText_SetText(objtext3b,rtos("00",GetAreaCommonData("History","Spell History",[])[SpellID][1]));
				}
			}
```
This code section is the culprit, chiefly the retrieval of the area common data for the stage history. Will continue investigating why.

**Comment by ExPorygon** [comment]
I've discovered the cause. The "Spell History" part of the "History" common data area does not exist. The correct common data are "Stage History" and "Practice History". I can only assume that it somehow still recognized the "Spell History" as valid and tried checking for values in a nonexistant array.

**Comment by ExPorygon** [comment]
Fixed in 648c7e8e81238f58d2510b97c1449434011f2e8c.
